rocket/src/FPU.scala

```diff
@@ -50,7 +48,7 @@ trait HasFPUCtrlSigs {
 
 class FPUCtrlSigs extends Bundle with HasFPUCtrlSigs
 
-class FPUDecoder(implicit p: Parameters) extends FPUModule()(p) {
+class FPUDecoder(xLen: Int, p: FPUParams) extends FPUModule(xLen, p) {
```

Comment:
Don't use p for FPUParams
```suggestion
class FPUDecoder(xLen: Int, fpuParams:  FPUParams) extends FPUModule(xLen, fpuParams) {
```